### PR TITLE
checkpoints: Add list command

### PR DESCRIPTION
**Pull request by nirs**
Add new command to list checkpoints. Similar to the purge command, it
can list checkpoints older than specified days.
    
This command is useful for managing checkpoints, and an example how to
find checkpoints and use checkpoint properties.

### CODE REVIEW

File: examples/checkpoints.py [hunk 1]
```diff
@@ -77,6 +100,29 @@ def main():
...
+
+        checkpoints = [
+            {
+                "id": checkpoint.id,
+                "creation_date": str(checkpoint.creation_date),
+                "state": str(checkpoint.state),
+                "description": checkpoint.description,
+            }
+            for checkpoint in checkpoints_service.list()
+            if args.days is None or age(checkpoint, now) > args.days
```
[the remark this answers is not in the document]
Comment (nirs) [reply examples/checkpoints.py:120]: This is the existing behavior in the purge command. The purpose of this change is
to add a new command, not to change the behavior of the --days argument.

Feel free to send a patch changing the behavior of the --days argument.

File: examples/checkpoints.py [hunk 2]
```diff
@@ -111,10 +157,9 @@ def cmd_purge(args):
         now = datetime.datetime.now(datetime.timezone.utc)
 
         for checkpoint in checkpoints_service.list():
-            checkpoint_age = now - checkpoint.creation_date
-
-            if checkpoint_age.days > args.days:
-                progress(f"Removing checkpoint {checkpoint.id}, created {checkpoint_age.days} days ago")
+            days = age(checkpoint, now)
```
[the remark this answers is not in the document]
Comment (nirs) [reply examples/checkpoints.py:160]: age_in_days() is little clumsy. How about days_old()?

File: examples/checkpoints.py [hunk 3]
```diff
@@ -136,5 +181,10 @@ def remove_checkpoint(checkpoint_service, timeout=60):
         time.sleep(1)
 
 
+def age(checkpoint, now):
```
[the remark this answers is not in the document]
Comment (nirs) [reply examples/checkpoints.py:184]: Exactly, we want to consider the same now value for all backups.

File: examples/checkpoints.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (bennyz) [reply examples/checkpoints.py:160]: I like personally like `days`

**Review by bennyz — APPROVED** [review]

File: examples/checkpoints.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (barpavel) [reply examples/checkpoints.py:120]: `So 1 day and 10 minutes is consider as 1 day.` - that was exactly my point. If user wanted checkpoint more than 1 day, the moment `age(checkpoint, now)` returns 1, this checkpoint is either 1 days' old exactly (0 hours, 0 minutes, 0 seconds and milliseconds, which is obviously little probability) or more than 1 day (i.e., 1 day and an hour). If I'm not mistaken, in the current implementation we'll return checkpoints at least 2 days' old.

File: examples/checkpoints.py [hunk 4]
```diff
@@ -77,6 +100,29 @@ def main():
     args.command(args)
 
 
+def cmd_list(args):
+    connection = common.create_connection(args)
```
[the remark this answers is not in the document]
Comment (nirs) [reply examples/checkpoints.py:104]: This breaks the json output. Can work if we move progress to stderr, but currently we use stdout.

File: examples/checkpoints.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (barpavel) [reply examples/checkpoints.py:120]: No problem, you already have my approval :)

File: examples/checkpoints.py [hunk 5]
```diff
@@ -77,6 +100,29 @@ def main():
...
+                "id": checkpoint.id,
+                "creation_date": str(checkpoint.creation_date),
+                "state": str(checkpoint.state),
+                "description": checkpoint.description,
+            }
+            for checkpoint in checkpoints_service.list()
+            if args.days is None or age(checkpoint, now) > args.days
+        ]
+
+    print(json.dumps(checkpoints, indent=2))
```
[the remark this answers is not in the document]
Comment (barpavel) [reply examples/checkpoints.py:123]: Ack.

File: examples/checkpoints.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (barpavel) [reply examples/checkpoints.py:104]: Ack.

**Review by barpavel — APPROVED** [review]
LGTM

**Review by bennyz — APPROVED** [review]

**Comment by nirs** [comment]
@oliel @mnecas can we merge?

**Review by oliel — APPROVED** [review]

**Pull request merged**